inc/Remote/XmlRpcServer.php

```diff
@@ -36,13 +37,13 @@ public function call($methodname, $args)
         } catch (AccessDeniedException $e) {
             if (!isset($_SERVER['REMOTE_USER'])) {
                 http_status(401);
-                return new ServerException("server error. not authorized to call method $methodname", -32603);
+                return new Error(-32603, "server error. not authorized to call method $methodname");
```

Comment:
I’m wondering if in the unit tests, client requests are already covered. The client included with DokuWiki could be used to do such requests. (See second example of the php clients, https://www.dokuwiki.org/devel:xmlrpc:clients#sample_php_client)
Test for these are welcome :)

Reply:
I used that client to debug the XmlRpcServer in integration tests i've written in cypress. But I didn't find a way to mock http requests inside phpunit as I'm used to in symfony so I let that out. Any idea from your side how to accomplish that?

Reply:
Ah, I’m not that much in the details there. There are some core testing functions: https://github.com/splitbrain/dokuwiki/tree/master/_test/core 
I expect that @splitbrain or @ssahara could give better help here, could you provide advice?

Reply:
If I understand correct, Klap-In reqested to create another unittest file likely `tests/inc/httpclient_xmlrpc.test.php`, in which there will be a code `new xmlrpcmsg('dokuwiki.createUser')`, as shown in second example of the php clients, [Samples of Clients using the XML-RPC Api](https://www.dokuwiki.org/devel:xmlrpc:clients#sample_php_client) ? I think likely no need to change inc/Remote/XmlRpcServer.php file.

Reply:
I would prefer to keep pull requests simple. If I understand it correctly, you're tackling a completely new problem here. Namely that Exceptions are not properly signalled to the client. Your solution seems sensible and having tests for that would be nice, but I would prefer both to be in a completely separate Pull Request. Let's keep this one strictly focused on the user creation.

Reply:
I reverted that change an will do a completely seperate Pull Request